File: app/src/main/java/com/alfianyusufabdullah/chatyuk/adapter/HolderChat.java
package com.alfianyusufabdullah.chatyuk.adapter;

import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.alfianyusufabdullah.chatyuk.MyApplication;
import com.alfianyusufabdullah.chatyuk.R;
import com.alfianyusufabdullah.chatyuk.model.ModelChat;
import com.alfianyusufabdullah.chatyuk.utils.Constan;

import java.util.List;

import butterknife.BindViews;
import butterknife.ButterKnife;

/**
 * Created by jonesrandom on 12/26/17.
 *
 * @site www.androidexample.web.id
 * @github @alfianyusufabdullah
 */

public class HolderChat extends RecyclerView.ViewHolder {

    @BindViews({R.id.cardFrom, R.id.cardTo})
    List<CardView> cardChat;

    @BindViews({R.id.fromUsername, R.id.toUsername})
    List<TextView> username;

    @BindViews({R.id.fromMessage, R.id.toMessage})
    List<TextView> message;

    @BindViews({R.id.fromTime, R.id.toTime})
    List<TextView> time;

    public HolderChat(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public void setContent(ModelChat chat) {

        String User = chat.getUser();
        String user = MyApplication.getInstance().getSharedPreferences().getString(Constan.PREF_USERNAME, "user");

        int POS_FROM = 0;
        int POS_TO = 1;

        if (User.equals(user)) {

            cardChat.get(POS_TO).setVisibility(View.VISIBLE);
            cardChat.get(POS_FROM).setVisibility(View.GONE);

            username.get(POS_TO).setText(chat.getUser());
            message.get(POS_TO).setText(chat.getMessage());
            time.get(POS_TO).setText(chat.getTime());

        } else {

            cardChat.get(POS_TO).setVisibility(View.GONE);
            cardChat.get(POS_FROM).setVisibility(View.VISIBLE);

            username.get(POS_FROM).setText(chat.getUser());
            message.get(POS_FROM).setText(chat.getMessage());
            time.get(POS_FROM).setText(chat.getTime());
        }
    }
}

File: app/src/main/java/com/alfianyusufabdullah/chatyuk/adapter/HolderChat.kt
package com.alfianyusufabdullah.chatyuk.adapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.alfianyusufabdullah.chatyuk.model.ModelChat
import com.alfianyusufabdullah.chatyuk.utils.ChatPreferences
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.row_chat.*

/**
 * Created by jonesrandom on 12/26/17.
 *
 * @site www.androidexample.web.id
 * @github @alfianyusufabdullah
 */

class HolderChat(override val containerView: View) : RecyclerView.ViewHolder(containerView) ,LayoutContainer {

    fun setContent(chat: ModelChat) {

        val from = chat.user
        val user = ChatPreferences.initPreferences(itemView.context).userInfo.username

        if (from == user) {
            cardTo.visibility = View.VISIBLE
            cardFrom.visibility = View.GONE

            toUsername.text = chat.user
            toMessage.text = chat.message
            toTime.text = chat.time

        } else {
            cardTo.visibility = View.GONE
            cardFrom.visibility = View.VISIBLE

            fromUsername.text = chat.user
            fromMessage.text = chat.message
            fromTime.text = chat.time
        }
    }
}